From 55b9c1c1dd4a80c179db4694e961fde043f078d6 Mon Sep 17 00:00:00 2001
From: fanzha3x <fanx.zhang@intel.com>
Date: Fri, 10 Nov 2017 10:10:22 +0800
Subject: [PATCH 2/2] Change activity startup function to
 "startActivityForResult" to return data.

While system execute command "am start -W -a android.intent.action.MAIN -c android.intent.category.HOME", It will wait activity to return data.
But system could not obtain any data if we use "startActivity" to start activity.
If we use "startActivityForResult" to start activity,the data will return immediately.

Change-Id: I2dbecfbf4641ebf5d8e9a7859a496274d038e307
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-52477
Signed-off-by: fanzha3x <fanx.zhang@intel.com>
---
 .../android/support/car/lenspicker/LensPickerTrampolineActivity.java  | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/support/car/lenspicker/LensPickerTrampolineActivity.java b/src/com/android/support/car/lenspicker/LensPickerTrampolineActivity.java
index 6a39c45..f73626f 100644
--- a/src/com/android/support/car/lenspicker/LensPickerTrampolineActivity.java
+++ b/src/com/android/support/car/lenspicker/LensPickerTrampolineActivity.java
@@ -37,6 +37,8 @@ import java.net.URISyntaxException;
 public class LensPickerTrampolineActivity extends Activity {
     private static final String TAG = "LensPickerTrampoline";
 
+    private static final int DEFAULT_ACTIVITY_REQUEST_CODE = 1;
+
     /**
      * A lock to be used when retrieving the {@link IStaturBarService}.
      */
@@ -140,7 +142,7 @@ public class LensPickerTrampolineActivity extends Activity {
         Intent intent = new Intent();
         intent.setComponent(defaultComponent);
 
-        startActivity(intent);
+        startActivityForResult(intent, DEFAULT_ACTIVITY_REQUEST_CODE);
     }
 
     /**
-- 
1.9.1

